### ISSUE TITLE
Empty cf. pointers; another example of lack of validation / consistency checks

### ISSUE THREAD

**Issue by m-linssen** [KELLIA/dictionary#157]
It seems like you've become stuck in this phase of the product, slowly getting it from infancy to maturity, now around the adolescence phase? Lacking procedures for this reocurring cycle of changes?

Given the increasing number of errors caused by changes to the first version, I think it is time to consider a strategy where you prioritise maintaining existing functionality over creating new one, and start by making an inventory of exactly all that you have: lists of cf. pointers for instance, that you cross-check to lists of lemma, that you cross-check to dictionary entries in their three languages, which you validate for correct spelling - and so on

Your product needs to move in cycles, releases, where you develop and unit test, and then "hand it over" to a different team in order to ensure unbiased and objective criticisim.
What I see now are errors in all kinds of different areas, likely as a result of making changes without cascading them down to their very last dependencies

You have, in C716 (ⲉⲣϣⲁⲛ):
cf | Partikel des Conditionalis | #ϣⲁⲛ

(I can assure you that Conditionalis is not German - that on a side note)

There are some 4-500 "cf. items", and 53 of those were already invalid in v 1.2019, meaning that they point to non-existing entries, that are not binary dictionary entries (ⲁⲟⲩⲱⲛ now is ⲁⲩⲱⲛ, for instance):

ⲁⲟⲩⲱⲛ
ⲁⲧ
ⲁϩ(ⲁ)-
ⲁϩⲏⲩ
ⲉⲓⲁ-
ⲉⲓⲁⲧ⸗
ⲉⲓⲉⲡ
ⲉϩⲁ-
ⲗⲁϭⲉ
ⲙ(ⲙ)ⲛ
ⲙⲁⲛⲉ-
ⲙⲃⲟⲕⲓ
ⲙⲉ-
ⲙⲛⲧⲉⲓⲁⲛϭⲁⲩⲟⲛ
ⲛⲁⲁ
ⲛⲓ-
ⲛ̄ⲧ-
ⲟⲩⲟϩ
ⲡⲓ-
ⲣⲉϥⲣⲓϥ
ⲣⲙ(ⲛ)
ⲣⲙⲃⲉⲕⲉ
ⲣⲟϥⲣⲉϥ
ⲣⲡⲗϭⲉ
ⲥⲁ-
ⲥⲁⲁϣϥ
ⲥⲁⲣ-
ⲥⲟⲩ
ⲥϯⲁⲛ
ⲧⲁⲓⲟ
ⲧⲁⲥⲟ
ⲧⲙⲁⲥⲓⲟ
ⲧⲥⲁⲉⲓⲟ
ⲧⲥⲉⲓⲟ
ⲧⲥⲛⲕⲟ
ⲧⲱϭⲉ
ⲱⲛϧ
ϣ
ϣⲁⲣ
ϣⲁⲥⲃⲁⲙⲡⲓ
ϣⲉⲃⲓⲱ
ϣⲉⲗϣⲏⲓⲗⲓ
ϣⲓϣⲓ
ϣⲓϯ
ϣⲛ-
ϥⲁⲓ-
ϫⲓ ⲉ-
ϫⲱ(ⲱ)ⲉ
ϫⲱⲃ
ϭⲁⲥⲓϩⲑⲟ
ϭⲓⲛ
ϭⲟⲥ-
ϭⲱⲙⲁ

Some of those have been fixed in v1.2, yet some of those have been broken in v1.2.
It is time that you start wondering whether you have thought this entire project through, because it looks like you're not in control and just making changes as you seem fit, without making sure that nothing gets broken elsewehere. I am sure that it is not your intention to be sloppy, but your CDO lacks accuracy in an increasing (and rather worrying) way

Some of the entries above have already been fixed in the current online version even though I retrieved them from v1.2

But the underlying issue is that your search function is relatively generous (for reasons that are valid within its own context) yet now returns 6 results for ⲙⲉ- (which now shows as ⲙⲉ in the online CDO versus the v1.2 XML)

Go to https://coptic-dictionary.org/entry.cgi?tla=C5497 (pearl), then look at 

cf. ⲙⲉ Wahrheit

Click the ⲙⲉ in there and you'll get these 6 results:

You searched for 'ⲙⲉ'
6 Results
ⲙⲁ-	ⲙⲁ⸗	if, when
ⲙⲉ	ⲙⲉⲓ	love
ⲙⲉ	--	love
ⲙⲉ	ⲙⲉⲉ	truth, justice, true person
ⲙⲉ	ⲙⲉⲉ	truthful, righteous, real, genuine
ⲙⲉⲛ	--	indeed, yes, yet, nonetheless, now, but (adversative), while, now ... and, but (introducing members of a set, without particle for contrasting members (e.g., δέ))

My German is fine so I know that Wahrheit means truth - but all this is turning out to be a labyrinth of your own making

If you need expertise and experience, I offer my help free of charge. It is not that complicated, just a bit complex - and complexity is fine, and consists of rules, and those can be written down, and automated, molded into procedures and working instructions that can be followed

But please, do fix your own growing legacy of leftovers and errors, and do it now

Regards,

Martijn Linssen

**Comment by dwerning** [comment]
Comment on the case of 
> cf. ⲙⲉ Wahrheit

Unfortunately, the raw data is not more precise than that
```
<xr type="cf">
   <ref target="#ⲙⲉ-">Wahrheit</ref>
</xr>
```
(BTW: The "#" does not seem to be fully appropriate since there is no proper xml target id "ⲙⲉ-", cf. https://tei-c.org/release/doc/tei-p5-doc/de/html/ref-att.pointing.html.)

Consequentely the webpage does nothing else but lanching a normal search https://coptic-dictionary.org/results.cgi?coptic=ⲙⲉ.
It is desireable to have a pointer to the actual entry C1789 in CDO, i.e.
```
<xr type="cf">
   <ref target="#C1789">ⲙⲉ Wahrheit</ref>
</xr>
```

**Comment by amir-zeldes** [comment]
@dwerning yes, that's exactly right - because the target attribute values are not actually XML IDs, the interface can only use the fallback strategy of searching for that item. It would be great if the targets were disambiguated IDs, and I'd be happy to support it in the interface it that were done, though I should point out that some aspects of the XML data might then not work as expected. 

For example, since gendered items have two entries following the TLA model for older Egyptian, a cross reference to, say, ϩⲙϩⲁⲗ, would need to choose one of the gendered forms (the first one would be the feminine, since we sort F>M). Similarly for cross-ref items that have both a noun and a verb from the same stem which look identical, only one of the forms could be chosen as a unique target.

The alternatives in such cases are to either define a policy on which item to choose, or to allow multiple target IDs and open a search page showing exactly those items.
